.circleci/cimodel/pytorch_build_definitions.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env python3
+
+from collections import OrderedDict
+
+import cimodel.miniutils as miniutils
+import cimodel.dimensions as dimensions
+import cimodel.conf_tree as conf_tree
+from cimodel.conf_tree import ConfigNode
+import cimodel.visualization as visualization
+
+
+DOCKER_IMAGE_PATH_BASE = "308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/"
+
+DOCKER_IMAGE_VERSION = 282
+
+
+class DockerHide(object):
+    """
+    Used for hiding name elements for construction of the Docker image path.
+    Name elements that are wrapped in this object may be part of the build configuration name, but
+    shall be excluded from the Docker path.
+    """
+    def __init__(self, val):
+        self.val = val
+
+    def __str__(self):
+        return self.val
+
+
+class Conf(object):
+    def __init__(self,
+                 distro,
+                 parms,
+                 pyver=None,
+                 cuda_version=None,
+                 is_xla=False,
+                 restrict_phases=None,
+                 gpu_resource=None,
+                 dependent_tests=None,
+                 parent_build=None):
+
+        self.distro = distro
+        self.pyver = pyver
+        self.parms = parms
+        self.cuda_version = cuda_version
+        self.is_xla = is_xla
+        self.restrict_phases = restrict_phases
+        self.gpu_resource = gpu_resource
+        self.dependent_tests = dependent_tests or []
+        self.parent_build = parent_build
+
+    def get_parms(self):
+        leading = ["pytorch"]
+        if self.is_xla:
+            leading.append(DockerHide("xla"))
+
+        cuda_parms = []
+        if self.cuda_version:
+            cuda_parms.extend(["cuda" + self.cuda_version, "cudnn7"])
+        return leading + ["linux", self.distro] + cuda_parms + self.parms
+
+    # TODO: Eliminate this special casing in docker paths
```

Comment:
When we begin merging caffe2 and pytorch jobs it will be convenient to have many more special cases in docker paths, until the dockers are finally merged.